front/app/services/permissions/rules/routePermissions.ts

```diff
@@ -22,22 +23,27 @@ export const MODERATOR_ROUTES = [
   '/admin/insights',
 ];
 
-export const isModeratorRoute = (item: IRouteItem) => {
-  return MODERATOR_ROUTES.includes(item.path);
+export const isModeratorIndexRoute = (item: IRouteItem) => {
+  return MODERATOR_INDEX_ROUTES.includes(item.path);
 };
 
 export const isModeratedProjectRoute = (
   item: IRouteItem,
   user: IUser | null
 ) => {
-  const idRegexp = /^\/admin\/projects\/([a-z0-9-]+)\//;
-  const matches = idRegexp.exec(item.path);
+  const { pathname: pathNameWithoutLocale } = removeLocale(item.path);
+  const idRegexp = /^\/admin\/projects\/([a-z0-9-]+)\/?/;
+  const matches = pathNameWithoutLocale
+    ? idRegexp.exec(pathNameWithoutLocale)
+    : false;
   const pathProjectId = matches && matches[1];
-  return (pathProjectId && isProjectModerator(user, pathProjectId)) || false;
+  return pathProjectId ? isProjectModerator(user, pathProjectId) : false;
 };
 
 export const isAdminRoute = (path: string) => {
-  return /^\/admin/.test(path);
+  const { pathname: pathNameWithoutLocale } = removeLocale(path);
```

Comment:
I don't think we want to make such a big change here to fix this one bug. This part of the codebase likely does need to be completely revamped, but other than the folder bug we found above, seems to be working fine for most of the app, and no other bugs have been reported.

This might technically be the right change but I don't think we want to introduce a relatively fundamental change to the permissions checks for the whole admin area of the app to fix this bug, it should come as part of a larger effort to rework this part of the codebase (and with the testing implied there)

Reply:
isAdminRoute is only used in 2 places. And all the places where I used this pathNameWithoutLocale are in the folder module.

Reply:
The second place is the permissions check that fires every on every route change in the admin part of the app:

`definePermissionRule('route', 'access', canAccessRoute);` which includes `isAdminRoute`. So it's being called a lot, and other than the bug we're fixing here, already works. So introducing a big change to it seems like there's a lot of potential for side effects, and isn't necessary to fix the specific bug we're looking at, which is why I don't think we should do it here.

Again, this is probably a fine change in general, but would require both automated and manual testing to make sure it doesn't have any unintended side effects in the rest of the admin section, so I think it's unnecessary to fix the bug here